src/wcmCommon.c

```diff
@@ -1843,27 +1852,50 @@ TEST_CASE(test_rebase_pressure)
 {
 	WacomDeviceRec priv = {0};
 	WacomDeviceRec base = {0};
+	WacomCommonRec common = {0};
 	WacomDeviceState ds = {0};
 	int pressure;
 
-	priv.minPressure = 4;
-	ds.pressure = 10;
+	priv.common = &common;
+	common.wcmMaxZ = 2000;
 
 	/* Pressure in out-of-proximity means get new preloaded pressure */
 	priv.oldState.proximity = 0;
 
+	/* pressure < pressure_threshold */
+	ds.pressure = 10;
+
 	/* make sure we don't touch priv, not really needed, the compiler should
 	 * honor the consts but... */
 	base = priv;
 
 	pressure = rebasePressure(&priv, &ds);
-	assert(pressure == ds.pressure);
+	assert(pressure > ds.pressure);
+	assert(memcmp(&priv, &base, sizeof(priv)) == 0);
+
+	/* pre-registered pressure > pressure_threshold */
+	ds.pressure = 100;
+
+	base = priv;
 
+	pressure = rebasePressure(&priv, &ds);
+	assert(pressure == ds.pressure);
 	assert(memcmp(&priv, &base, sizeof(priv)) == 0);
 
 	/* Pressure in-proximity means rebase to new minimum */
 	priv.oldState.proximity = 1;
 
+	priv.minPressure = 20;
+	ds.pressure = 10;
+
+	base = priv;
+
+	pressure = rebasePressure(&priv, &ds);
+	assert(pressure == priv.minPressure);
+	assert(memcmp(&priv, &base, sizeof(priv)) == 0);
+
+	ds.pressure = 100;
```

Comment:
rebasepressure is to report "the minimum pressure based on the current minimum pressure and the hardware state". Below the min_pressure_threshold is a hardware state, which means the value should be filtered.  The minimum pressure doesn't imply 0 pressure. So rebasepressure is basically to find the smallest pressure we can based on. The added code didn't change that description. It changed your expectation though.

Reply:
Maybe we should rename `rebasePressure` to `calculateMinPressure` or something instead? I don't remember why I picked "rebase" here - maybe it did the actual rebasing in some earlier version and then got stripped down to just returning the minimum?

Reply:
Ok, I think I know why you were expecting min_pressure as 0. Refer to [https://github.com/linuxwacom/xf86-input-wacom/blob/master/src/wcmCommon.c#L1420](url). With the new code, we need to set priv->maxRawPressure = min_pressure_threshold instead of 0, refer to lines 1422 and 1441. So, rebasepressure isn't a bad name since it is used to set the base/minimal pressure for normalizePressure. We just need to be careful that we update all the other relevant code as well.

Reply:
tbh, my head is spinning now from so many different variables that are all slightly different `wcmMaxZ`, `maxRawPressure`, then we have `detectPressureIssue` separate from `setPressureButton` etc. But yes, I agree, somewhere in this section should accommodate for the `min_pressure` that we have now, but don't ask me where without me spending some more time to figure all this out :)

Reply:
I am thinking if kernel is a better place to address this issue, which covers everything in userland. What do you think?